.github/workflows/CI.yml

```diff
@@ -36,3 +36,7 @@ jobs:
       run: cargo build --verbose
     - name: Run tests
       run: cargo test --verbose
+    - name: Run rustfmt
+      run: cargo fmt -- --check
+    - name: Run clippy
+      run: cargo clippy --all-targets --all-features -- -D warnings
```

Comment:
`-D warnings` is a bit harsh I feel haha. I'd rather enable the feature where it logs warnings and it shows up when browsing the code on GitHub, and keep the warnings as such to not block PRs.

Reply:
Ah, also, I think it's better to run clippy before build, so that it fails early in case of a compile or lint error.

Reply:
> I'd rather enable the feature where it logs warnings and it shows up when browsing the code on GitHub, and keep the warnings as such to not block PRs.

Is that a clippy feature? I don't think I've seen this before so I'm not sure I understand what you mean.

Reply:
It's from this action: https://github.com/clechasseur/rs-clippy-check

Reply:
> It's from this action: https://github.com/clechasseur/rs-clippy-check

Ah I see! Do you know if this requires nightly or can it run on stable?

Reply:
Also just notice we're using 1.80 instead of 1.85 so I'll update that.

Reply:
On that note, it might be good to run on the latest 3 versions of Rust, so 1.83, 1.84, and 1.85 (like I do in [clogbox](https://github.com/SolarLiner/clogbox/blob/f4bf1e26deaebfa669df7850435a449d9e7da3cf/.github/workflows/test.yaml#L14))